app/workers/scheduled_publishing_worker.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class ScheduledPublishingWorker
+  include Sidekiq::Worker
+  sidekiq_options retry: 10
+  sidekiq_retry_in { 30 }
+
+  def perform(id)
+    Edition.transaction do
+      edition = Edition.lock.find(id)
+
+      unless edition.scheduled?
+        raise "Cannot schedule an edition that is not in a scheduled state"
+      end
+
+      if scheduled_publishing_datetime_in_the_past?(edition)
+        raise "Cannot schedule an edition whose scheduled publishing datetime is in the past"
+      end
+
+      user = edition.status.created_by
+      reviewed = edition.status.details&.reviewed
+
+      PublishService.new(edition.document)
+                    .publish(user: user, with_review: reviewed)
+    end
+  end
+
+private
+
+  def scheduled_publishing_datetime_in_the_past?(edition)
+    edition.scheduled_publishing_datetime <= Time.zone.now
```

Comment:
super duper minor, but using `Time.current` is generally preferred to `Time.zone.now` since it's not nested. 

There was this a bit ago: https://github.com/alphagov/content-publisher/pull/708